File: evo_motion/view/camera.py
from abc import ABC, abstractmethod

from glm import vec3


class Camera(ABC):
    @abstractmethod
    def pos(self) -> vec3:
        pass

    @abstractmethod
    def up(self) -> vec3:
        pass

    @abstractmethod
    def look_vec(self) -> vec3:
        pass


class StaticCamera(Camera):
    def __init__(
        self,
        pos: vec3,
        look_vec: vec3,
        up: vec3,
    ):
        self.__pos = pos
        self.__look_vec = look_vec
        self.__up = up

    def pos(self) -> vec3:
        return self.__pos

    def up(self) -> vec3:
        return self.__up

    def look_vec(self) -> vec3:
        return self.__look_vec

File: src/view/camera.cpp
//
// Created by samuel on 17/12/22.
//

#include "camera.h"

StaticCamera::StaticCamera(glm::vec3 pos, glm::vec3 look, glm::vec3 up) : pos_vec(pos), look_vec(look), up_vec(up) {

}

glm::vec3 StaticCamera::pos() {
    return pos_vec;
}

glm::vec3 StaticCamera::look() {
    return look_vec;
}

glm::vec3 StaticCamera::up() {
    return up_vec;
}
